ui/component/comment/view.jsx

```diff
@@ -28,54 +40,166 @@ function Comment(props: Props) {
     }
-  }, [isResolvingUri, shouldFetch, author, authorUri, resolveUri]);
+
+    if (isEditing) {
+      setCharCount(editedMessage.length);
+
+      // a user will try and press the escape key to cancel editing their comment
+      const handleEscape = (event) => {
+        if (event.keyCode === ESCAPE_KEY) {
+          setEditing(false);
+        }
+      };
+
+      window.addEventListener('keydown', handleEscape);
+
+      // removes the listener so it doesn't cause problems elsewhere in the app
+      return () => {
+        window.removeEventListener('keydown', handleEscape);
+      };
+    }
+  }, [isResolvingUri, shouldFetch, author, authorUri, resolveUri, editedMessage, isEditing, setEditing]);
+
+  // edit button calls this function
+  function handleSetEditing() {
+    setEditing(true);
+  }
+
+  // when the body is changed, form calls this function
+  function handleEditMessage(event) {
+    setCommentValue(event.target.value);
+  }
+
+  // when the user is done, this function is called
+  function handleSubmit() {
+    updateComment(commentId, editedMessage);
+    setCurrentMessage(editedMessage);
+    setEditing(false);
+  }
+
+  function handleDeleteComment() {
+    openModal(MODALS.DELETE_COMMENT, {
+      deleteComment: () => {
+        deleteComment(commentId);
+      },
+    });
+  }
+
+  function handleMouseOver() {
+    setMouseHover(true);
+  }
+
+  function handleMouseOut() {
+    setMouseHover(false);
+  }
 
   return (
-    <li className="comment">
+    <li className="comment" onMouseOver={handleMouseOver} onMouseOut={handleMouseOut}>
       <div className="comment__author-thumbnail">
         {authorUri ? <ChannelThumbnail uri={authorUri} obscure={channelIsBlocked} small /> : <ChannelThumbnail small />}
       </div>
 
       <div className="comment__body_container">
-        <span className="comment__meta">
-          {!author ? (
-            <span className="comment__author">{__('Anonymous')}</span>
-          ) : (
-            <Button
-              className="button--uri-indicator truncated-text comment__author"
-              navigate={authorUri}
-              label={author}
-            />
-          )}
-
-          <time className="comment__time" dateTime={timePosted}>
-            {relativeDate(timePosted)}
-          </time>
-        </span>
+        <div className="comment__meta">
+          <div className="comment__meta-information">
+            {!author ? (
+              <span className="comment__author">{__('Anonymous')}</span>
+            ) : (
+              <Button
+                className="button--uri-indicator truncated-text comment__author"
+                navigate={authorUri}
+                label={author}
+              />
+            )}
+            <time className="comment__time" dateTime={timePosted}>
+              {relativeDate(timePosted)}
+            </time>
+          </div>
+          <div className="comment__meta-menu">
+            { commentIsMine && (
```

Comment:
The space in between `{` and `commentIsMine` is because you disabled the precommit hook which formats the code.